src/groupeddataframe/grouping.jl

```diff
@@ -181,6 +197,242 @@ end
 Base.getindex(gd::GroupedDataFrame, idxs::Colon) =
     GroupedDataFrame(gd.parent, gd.cols, gd.groups, gd.idx, gd.starts, gd.ends)
 
+
+# Eltypes of the grouping columns
+_grouptypes(gd::GroupedDataFrame) = Tuple(eltype(gd.parent[!, c]) for c in gd.cols)
+
+# Iterator over values of grouping columns for each group
+function _groupvalues(gd::GroupedDataFrame)
+    T = Tuple{_grouptypes(gd)...}
+    return (T(gd.parent[gd.idx[s], gd.cols]) for s in gd.starts)
+end
+
+# Get values of grouping columns for single group by its index
+_groupvalues(gd::GroupedDataFrame, i::Integer) = Tuple(gd.parent[gd.idx[gd.starts[i]], gd.cols])
```

Comment:
just to be sure all will work correctly in the future - maybe add another method for `i::Bool` that throws an error?

Reply:
Ah - never mind - it will be a problem only on Julia 1.0, so probably we can ignore it.